imports/client/components/AllProfileListPage.tsx

```diff
@@ -1,17 +1,25 @@
-import { _ } from 'meteor/underscore';
-import React from 'react';
-import PropTypes from 'prop-types';
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
 import { withTracker } from 'meteor/react-meteor-data';
 import { withBreadcrumb } from 'react-breadcrumbs-context';
 import subsCache from '../subsCache';
 import ProfileList from './ProfileList';
-import ProfileSchema from '../../lib/schemas/profiles';
+import ProfileSchema, { ProfileType } from '../../lib/schemas/profiles';
 import Profiles from '../../lib/models/profiles';
 
-class AllProfileListPage extends React.Component {
+interface AllProfileListPageProps {
+  ready: boolean;
+  profiles: ProfileType[];
+}
+
+class AllProfileListPage extends React.Component<AllProfileListPageProps> {
   static propTypes = {
     ready: PropTypes.bool.isRequired,
-    profiles: PropTypes.arrayOf(PropTypes.shape(ProfileSchema.asReactPropTypes())).isRequired,
+    profiles: PropTypes.arrayOf(
+      PropTypes.shape(
+        ProfileSchema.asReactPropTypes<ProfileType>()
+      ).isRequired as React.Validator<ProfileType>
```

Comment:
There's some wonkiness because of how the ambient type declarations for `PropTypes` works. It inspects to see if a field is optional (by seeing if its type includes either `null` or `undefined`), and if so, makes the property optional (in the `foo?: string`) sense. That doesn't match our schema type declarations, which accept `undefined` but "require" the property to be present.